projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -579,6 +579,44 @@ function wpcom_launchpad_get_task_definitions() {
 				return '/site-editor/' . $data['site_slug_encoded'];
 			},
 		),
+		'blogname_set'                       => array(
+			'get_title'            => function () {
+				return __( 'Give your site a name', 'jetpack-mu-wpcom' );
+			},
+			'is_complete_callback' => 'wpcom_launchpad_is_legacy_task_option_completed',
```

Comment:
Might it not be more robust to have the completion and/or visibility logic combine checks against the legacy option and the launchpad option? I am especially thinking of Atomic sites, where we won't have access to code in the `Onboarding` namespace. This specific task feels equivalent to the site title task, so we're already tracking things there.

Reply:
Oh! One of my pushes didn't go through yesterday, the `wpcom_launchpad_is_legacy_task_option_completed` should look like this:
```php
function wpcom_launchpad_is_legacy_task_option_completed( $task ) {
	$option_name = 'onboarding_checklist_task_' . $task['id'];
	return get_option( $option_name, false ) || wpcom_launchpad_is_task_option_completed( $task );
}
```